### PR TITLE
Workflow

### PR DESCRIPTION
**Pull request by abesto**
Based off of #15, probably a good idea to merge that first, and review only the remaining diff.

### CODE REVIEW

**Comment by abesto** [comment]
Oh, and, my new hobby is pinging @kbence

**Comment by abesto** [comment]
Full (2 hours long) integration test: https://prezi.slack.com/archives/prio1-2016-02-12-2

**Comment by kbence** [comment]
I'm already beyond Ballmer peak, so I postpone this if you don't mind... :)

**Comment by abesto** [comment]
Rebased to master after the merge of #15

File: pasha_modules/util.coffee [hunk 1]
```diff
@@ -217,29 +218,6 @@ pagerdutyAlert = (description) ->
...
-        try
-            nagTarget = if prio1.role.comm then prio1.role.comm else prio1.role.starter
-            message = "@#{getUser(nagTarget, null, state.users).name}, please use '#{constant.botName} status <some status update>' regularly, the last status update for the current outage was at #{moment.unix(prio1.time.lastStatus).fromNow()}"
-            if prio1.channel.name?
-                adapter.messageRoom prio1.channel.name, message
-            else
-                msg.send message
-        catch error
-            scribeLog "ERROR nagger #{error}"
-    naggerCallbackId = setInterval(nagger, 10 * 60 * 1000)
```
Comment (kbence) [inline pasha_modules/util.coffee:241]: Dat naming :).

File: pasha_modules/util.coffee [hunk 2]
```diff
@@ -255,7 +233,6 @@ module.exports = {
     sendConfirmEmail: sendConfirmEmail
     sendStatusEmail: sendStatusEmail
     pagerdutyAlert: pagerdutyAlert
-    startNag: startNag
```
Comment (kbence) [inline pasha_modules/util.coffee:258]: Btw. we might want to refactor this file in the future to use `@` names for public functions

Comment (zsol) [inline pasha_modules/util.coffee:258]: that slack link does not work for me ;)

File: scripts/pasha_main.coffee [hunk 3]
```diff
@@ -112,6 +120,109 @@ module.exports = (robot) ->
...
+        twoHours: () =>
+            @loadState()
+            @remind 'marketing', "The prio1 has been going on for two hours. It's time to work with our Brand Communications director to alert HOX."
+
+        stop: () =>
+            @loadState()
+            @remind 'support', 'If possible, please verify that the prio1 is over. If not possible, please acknowledge that you understand engineering believes the prio1 is over.'
+            @remind 'support', "Please work with @#{@state.prio.role.marketing} to update public communications channels, resolve the outstanding issue and update the green/yellow/red status on the status page. Ask @#{@state.prio1.role.comm} for clarifications as needed."
+            clearTimeout @nextTimeoutId if @nextTimeoutId?
+            clearInterval @tenMinuteIntervalId if @tenMinuteIntervalId?
```
Comment (kbence) [inline scripts/pasha_main.coffee:224]: Nice and neat. Only one thing: it feels a bit weird to have the next timeout defined in all step (eg. I know it's unlikely, but what happens if we redeploy pasha - or it just dies and have to be restarted - during an outage). I'd rather store the next event in brain and check it in - let's say - every minutes. But of course, I can live with this solution (it's way better than not having this :smile:)

**Comment by kbence** [comment]
Big like :+1:

File: scripts/pasha_main.coffee (hunk 3 above)
Comment (abesto) [inline scripts/pasha_main.coffee:224]: Totally agree, the first thing I tried to do before implementing this workflow was to change the old "every 10 minutes" nagger to a brain-based setup. I quickly reached the conclusion that it's more complexity than I wanted to deal with for now. On the upside, now that the workflow is implemented, we'll have a better idea of what use-cases a brain-based solution needs to support when / if we implement it in the future.

**Pull request merged**